packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -56,21 +55,21 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
 
     const addField = useCallback(
         (item: any = undefined) => {
-            append(item);
+            let defaultValue = item;
+            if (item == null) {
+                defaultValue = {} as Record<string, unknown>;
+                Children.forEach(children, input => {
+                    if (React.isValidElement(input)) {
+                        defaultValue[input.props.source] =
+                            input.props.defaultValue ?? '';
+                    }
+                });
+            }
+            append(defaultValue);
         },
-        [append]
+        [append, children]
     );
 
-    // add field and call the onClick event of the button passed as addButton prop
```

Comment:
Why did you remove this part?

Reply:
Because it was unnecessary and actually induced an issue by adding an item twice

Reply:
restored